huaweicloud/services/dli/resource_huaweicloud_dli_queue_v1.go

```diff
@@ -0,0 +1,229 @@
+const RESOURCE_MODE_SHARED, RESOURCE_MODE_EXCLUSIVE = 0, 1
+
+func ResourceDliQueueV1() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDliQueueCreate,
+		Read:   resourceDliQueueV1Read,
+		Delete: resourceDliQueueV1Delete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(string)
+					if !regexp4Name.MatchString(v) {
+						errs = append(errs, fmtp.Errorf("%q can contain only digits, lower letters, and underscores (_) ", key))
+					}
+					return
+				},
+			},
+
+			"queue_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Default:      "sql",
+				ValidateFunc: validation.StringInSlice([]string{"sql", "general"}, false),
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"cu_count": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntInSlice([]int{CU_16, CU_64, CU_256}),
+			},
+
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			"platform": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Default:      "x86_64",
+				ValidateFunc: validation.StringInSlice([]string{"x86_64", "aarch64"}, false),
+			},
+
+			"resource_mode": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntInSlice([]int{RESOURCE_MODE_SHARED, RESOURCE_MODE_EXCLUSIVE}),
+			},
+
+			"feature": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"basic", "ai"}, false),
+			},
+
+			"tags": {
+				Type: schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"create_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceDliQueueCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	dliClient, err := config.DliV1Client(config.GetRegion(d))
+	if err != nil {
+		return fmt.Errorf("creating dli client failed: %s", err)
+	}
+
+	queueName := d.Get("name").(string)
+
+	logp.Printf("[DEBUG] create dli queues queueName: %s", queueName)
+	createOpts := queues.CreateOpts{
+		QueueName:           queueName,
+		QueueType:           d.Get("queue_type").(string),
+		Description:         d.Get("description").(string),
+		CuCount:             d.Get("cu_count").(int),
+		EnterpriseProjectId: config.GetEnterpriseProjectID(d),
+		Platform:            d.Get("platform").(string),
+		ResourceMode:        d.Get("resource_mode").(int),
```

Comment:
The type of `ResourceMode` is 'int `json:"resource_mode"`', when the value is `0` this parameter will be discarded during conversion process of sending the http request to the server.

Reply:
hava change the json tag in sdk. Always serialize this field to zero when it's nil